### PR TITLE
Parametrize `shuffle` for `test_dataframe.py::test_shuffle`

### LINKED ISSUE #640: Parametrize `shuffle` for `test_dataframe.py::test_shuffle`

### PR DESCRIPTION
**Pull request by hendrikmakait**
* Closes #640

### CODE REVIEW

**Comment by hendrikmakait** [comment]
CI failure caused by PyArrow silently coercing non-string column names to `str` (https://github.com/coiled/coiled-runtime/actions/runs/3835692936/jobs/6529832882#step:5:745)

dask/distributed#7447 will now raise on non-string column names for a clearer error message.

File: alembic/versions/a97d9375430f_default_parameter_for_test_dataframe_py_.py
Comment (fjetter) [inline alembic/versions/a97d9375430f_default_parameter_for_test_dataframe_py_.py:14]: I guess this conflicts with https://github.com/coiled/coiled-runtime/pull/632 or can alembic resolve non-linear migrations?

**Comment by hendrikmakait** [comment]
Blocked until #644 has been merged, after which the migration needs to be adapted.

**Comment by hendrikmakait** [comment]
Serialized migrations with #658, ready for review.

**Review by ncclementi — APPROVED** [review]
LGTM

**Pull request merged**
